src/OpenSage.Game/Logic/Object/Draw/W3dModelDraw.cs

```diff
@@ -172,10 +172,15 @@ private W3dModelDrawConditionState CreateModelDrawConditionStateInstance(ModelCo
             {
                 foreach (var particleSysBone in conditionState.ParticleSysBones)
                 {
-                    var particleSystemTemplate = _contentManager.IniDataContext.FXParticleSystems.First(x => x.Name == particleSysBone.ParticleSystem);
+                    var particleSystemTemplate = _contentManager.IniDataContext.FXParticleSystems.FirstOrDefault(x => x.Name == particleSysBone.ParticleSystem);
                     if (particleSystemTemplate == null)
                     {
-                        particleSystemTemplate = _contentManager.IniDataContext.ParticleSystems.First(x => x.Name == particleSysBone.ParticleSystem).ToFXParticleSystemTemplate();
+                        particleSystemTemplate = _contentManager.IniDataContext.ParticleSystems.FirstOrDefault(x => x.Name == particleSysBone.ParticleSystem).ToFXParticleSystemTemplate();
+
+                        if (particleSystemTemplate == null)
```

Comment:
This will never be reached, because it will crash on the line above when calling the ToFXParticleSystemTemplate method, if the result of FirstOrDefault is null.